cluster-autoscaler/utils/deletetaint/delete_test.go

```diff
@@ -60,20 +93,57 @@ func TestCheckNodes(t *testing.T) {
 	assert.True(t, time.Now().Sub(*val) < 10*time.Second)
 }
 
+func TestSoftQueryNodes(t *testing.T) {
```

Comment:
nit: how is this scenario intended to be different from TestSoftMarkNodes, other than making a couple more assertions?

Reply:
First scenario tests Mark and Has (so baseline scenario), second tests Get. I could remove Has call to make it more obvious, but Has is a general assertion strategy in all of those tests.

Reply:
Also I have included additional basic scenario. TestMark tests with Mark, TestCheck tests with addTaintToSpec. Other tests uses addTaintToSpec as a building block so I felt that it would be beneficial to test it separately